scripts/renderers.py

```diff
@@ -99,14 +159,81 @@ def sorting_header(key):
         output.extend(header)
         return output
 
-    def render_csv_report(self, envs_ida_toggle_data, file_path="report.csv", toggle_types=None, header=None):
+    def combine_envs_data_under_toggle_identifier(self, envs_data, type_filter=None):
         """
-        takes data, processes it, and outputs it in csv form
+        envs data is structured: envs->ida->toggles, this converts to (toggle, ida, toggle_type)->{env1_toggle_data, env2_toggle_data}
```

Comment:
1. I'm unclear what `(toggle, ida, toggle_type)->{env1_toggle_data, env2_toggle_data}` means.
2. If we were to output this report in HTML, would it use this same data structure? Is it more a function of the report, and not of the format of the report?  If so, this should also be moved to wherever the processing code is moved (e.g. `ToggleReportDataProcessor`?).

```suggestion
        Flattens envs_data structure to a new data structure by toggle.

        Original envs_data structure: envs->ida->toggles
        New data structure: (toggle, ida, toggle_type)->{env1_toggle_data, env2_toggle_data}
```